src/core/validate-graph.ts

```diff
@@ -60,6 +60,17 @@ export function validatePackageURL(pkg: types.PkgInfo): void {
         );
         break;
 
+      // The PURL spec for Linux distros does not include the source in the name.
+      // This is why we relax the assertion here and match only on the package name:
+      // <source name>/<package name> - we omit the source name
+      // For now, make this exception only for deb to cover a support case.
+      case 'deb': {
+        const pkgName = pkg.name.split('/').pop();
+        const purlName = purlPkg.name.split('/').pop();
+        assert(pkgName === purlName, 'name and packageURL name do not match');
+        break;
```

Comment:
Again, I don't think the source should be part of the PURL Namespace? It should be in a qualifier.

Also, the `purlPkg.name.split('/').pop()` shouldn't be necessary - even if we encoded the `source` like `pkg:/deb/source/pkgName@version` (which is an invalid purl according to the spec!), `source` would be in the `namespace` field, not in the name.

Reply:
Fixed I think!